src/system_metrics/service_mongoose_system_metrics.erl

```diff
@@ -25,7 +25,8 @@
 
 -export([verify_if_configured/0]).
 
--record(system_metrics_state, {report_after, reporter_monitor = none, reporter_pid = none}).
+-record(system_metrics_state, {report_after, reporter_monitor = none,
+                               reporter_pid = none, prev_report = []}).
```

Comment:
I'd give this some type information, like
```erlang
-record(system_metrics_state,
        {
         report_after :: non_neg_integer(),
         reporter_monitor = none :: reference(),
         reporter_pid = none :: pid(),
         prev_report = []
        }).
```
And it itches me to leave `prev_report` untyped but ok, no need to export types from other modules and so on, this is sent as a message anyway so nor dialyzer nor gradualizer will do anything anyway :man_shrugging: